plugins/ImageGraph/API.php

```diff
@@ -129,404 +130,453 @@ public function get(
         $segment = false
     )
     {
-        Piwik::checkUserHasViewAccess($idSite);
-
-        // Health check - should we also test for GD2 only?
-        if (!SettingsServer::isGdExtensionEnabled()) {
-            throw new Exception('Error: To create graphs in Piwik, please enable GD php extension (with Freetype support) in php.ini,
-            and restart your web server.');
-        }
-
-        $useUnicodeFont = array(
-            'am', 'ar', 'el', 'fa', 'fi', 'he', 'ja', 'ka', 'ko', 'te', 'th', 'zh-cn', 'zh-tw',
-        );
-        $languageLoaded = Translate::getLanguageLoaded();
-        $font = self::getFontPath(self::DEFAULT_FONT);
-        if (in_array($languageLoaded, $useUnicodeFont)) {
-            $unicodeFontPath = self::getFontPath(self::UNICODE_FONT);
-            $font = file_exists($unicodeFontPath) ? $unicodeFontPath : $font;
-        }
-
-        // save original GET to reset after processing. Important for API-in-API-call
-        $savedGET = $_GET;
-
-        try {
-            $apiParameters = array();
-            if (!empty($idGoal)) {
-                $apiParameters = array('idGoal' => $idGoal);
+        $idSiteArr = array();
```

Comment:
Here you could use `$idSiteArr = Site::getIdSitesFromIdSitesString($idSite)`. This handles eg the case of `idSite=all` and does an `explode(',', $idSite)` etc